src/runtime/compact.cpp

```diff
@@ -60,11 +70,8 @@ object_offset object_compactor::to_offset(object * o) {
 }
 
 void object_compactor::insert_terminator(object * o) {
-    void * mem = alloc(sizeof(object) + sizeof(object*));
-    object * r = new (mem) object(object_kind::External, object_memory_kind::Region);
-    r->m_kind  = TERMINATOR_ID;
-    object** ptr = reinterpret_cast<object**>(reinterpret_cast<char*>(r) + sizeof(object));
-    *ptr = to_offset(o);
+    void * mem = alloc(sizeof(terminator_object));
+    new (mem) terminator_object(to_offset(o));
```

Comment:
Why do we need this modification?

Reply:
The old `ptr` was misaligned. Using a struct for the two datums implicitly fixes the alignment.

Reply:
I see, `sizeof(object)` was not a multiple of 8, right?

Reply:
Yes, exactly